src/weapon/w_050.c

```diff
@@ -192,9 +192,207 @@ void func_162000_8017BA38(Entity* ent, Point16* outPoint, bool arg2) {
 
 INCLUDE_ASM("weapon/nonmatchings/w_050", func_ptr_80170004);
 
-INCLUDE_ASM("weapon/nonmatchings/w_050", func_ptr_80170008);
+void func_ptr_80170008(Entity* self) {
+    const int PrimCount = 16;
+    Primitive* prim;
+    s16 angle;
+    s32 magnitude;
+    s32 i;
+    s16 offsetX, offsetY;
+    s16 baseX, baseY;
+    u8 color;
+
+    if (PLAYER.ext.player.anim <= 0x40 || PLAYER.ext.player.anim >= 0x48 ||
+        g_Player.unk46 == 0) {
+        DestroyEntity(self);
+        return;
+    }
+    offsetX = 0xC;
+    if (PLAYER.facingLeft) {
+        offsetX = -0xC;
+    }
+    self->posX.i.hi = PLAYER.posX.i.hi + offsetX;
+
+    offsetY = PLAYER.posY.i.hi;
+    self->posY.i.hi = offsetY - 0x1A;
+    if (PLAYER.drawFlags & FLAG_DRAW_ROTY) {
+        self->posY.i.hi = offsetY - 0x1F;
+    }
+
+    switch (self->step) {
+    case 0:
+        self->primIndex = g_api.AllocPrimitives(PRIM_LINE_G2, PrimCount);
+        if (self->primIndex != -1) {
```

Comment:
```suggestion
        if (self->primIndex != -1) {
```
Let's invert the condition to remove indentation: https://decomp.me/scratch/nm9HF

The `if (primIndex == -1) { DestroyEntity(self); return; }` is a pattern found in almost all the code base